psyneulink/core/llvm/codegen.py

```diff
@@ -49,12 +61,39 @@ def _exp(x):
             helpers.call_elementwise_operation(self.ctx, self.builder, x, helpers.exp, output_ptr)
             return output_ptr
 
-        self.register['sum'] = _vec_sum
+        def _max(x):
+            assert helpers.is_vector(x) or helpers.is_2d_matrix(x), "Attempted to call max on invalid variable! Only 1-d and 2-d lists are supported!"
+            curr = builder.alloca(ctx.float_ty)
+            has_set = False
```

Comment:
this requires the condition to be 
`greater = builder.not_(builder.fcmp_ordered('<', element, builder.load(curr)))`,
since ordered comparisons always return `False` is one of the operands is `NaN`.

Fixed by swapping to unordered comparison.